mlrun/datastore/sources.py

```diff
@@ -91,13 +91,28 @@ def filter_df_start_end_time(self, df, time_field):
             )
         return df
 
-    def to_spark_df(self, session, named_view=False, time_field=None):
-        if self.support_spark:
-            df = session.read.load(**self.get_spark_options())
-            if named_view:
-                df.createOrReplaceTempView(self.name)
-            return df
-        raise NotImplementedError()
+    def to_spark_df(self, session, named_view=False, time_field=None, columns=None):
+        if not self.support_spark:
+            raise NotImplementedError()
+
+        df = session.read.load(**self.get_spark_options())
+        if named_view:
+            df.createOrReplaceTempView(self.name)
+        return self._filter_spark_df(df, time_field, columns)
```

Comment:
`_ingest_with_spark` already does this filtering. Need to avoid filtering twice.